bullet_train/test/lib/tasks/bullet_train_tasks_test.rb

```diff
@@ -0,0 +1,51 @@
+require "test_helper"
+require "minitest/spec"
+require "rake"
+Rails.application.load_tasks
+
+describe "rake bullet_train:resolve --interactive" do
+  before do
+    @original_stdin = $stdin
+    @original_stdout = $stdout
+    $stdout = StringIO.new
+  end
+
+  after do
+    $stdin = @original_stdin
+    $stdout = @original_stdout
+  end
+
+  describe "when passed BEGIN absolute path from annotated view" do
+    it "resolves the path" do
+      inputs = [
+        "<!-- BEGIN /ruby-3.2.2/gems/bullet_train-themes-light-1.2.26/app/views/themes/light/_notices.html.erb -->",
```

Comment:
I don't want to rely on specific versions of Ruby or our gems for these tests, so can we somehow justify these with a regular expression, or grab the ruby version and current gem version?

```ruby
# Something like this for the Ruby version
File.open(".ruby-version").readlines.pop

# Something like this for the gem version
`bundle show bullet_train`.regexp_to_get_version_only
```

Reply:
I'm pretty sure I could strip that first part out. I actually did strip out some of the absolute path (my user account name). It's not a full absolute path. I'm not sure if that's a good idea. The _implementation_ ignores the good amount of the absolute path, but perhaps we should keep some kind of example absolute path nevertheless? 🤔 
For the sake of time, would it work if I removed ruby version as well since the absolute path doesn't have to be within the `ruby-<version>` directory. I'm using RVM so the gems are stored in `$HOME/.rvm/gems/ruby-<version>/gems`?

Reply:
> For the sake of time, would it work if I removed ruby version as well since the absolute path doesn't have to be within the ruby-<version> directory. I'm using RVM so the gems are stored in $HOME/.rvm/gems/ruby-<version>/gems?

Yes, I thought about this as well. Without prying too deep into it, I think we'll actually need to use an absolute path because we're running the test from inside `bullet_train-core/bullet_train` as opposed to a newly cloned repository itself where the task is usually run, but I'd have to check though.

If we go with an absolute path (which will most likely be more sustainable in the long run), it might take some work but you could potentially grab the path like this:
```
# Finds the installation directory for all the gems
> gem environment | grep "\- INSTALLATION DIRECTORY" | sed -E 's/- INSTALLATION DIRECTORY: //'

# You'd need to pull the version number and build the project path
# but this will at least give you the gem since `bundle show` won't work in bullet_train-core/bullet_train
> gem list | grep bullet_train-themes-light
#=> bullet_train-themes-light (1.2.26, 1.2.25, 1.2.24)
```

Then you could build the string like this:

```ruby
"<!-- BEGIN #{absolute_path} -->"
```